### PR TITLE
services/horizon/internal: Add support for CAP 15 in Horizon

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

The Stellar Core protocol 13 release will introduce a new type of transaction called Fee Bump transactions. Here is a summary of the changes (see https://github.com/stellar/stellar-protocol/blob/master/core/cap-0015.md for the full set of details):

The XDR representation of a transaction envelope has changed to be a union of legacy (pre protocol 13) transactions, normal transactions where the source account pays for their own transactions, and fee bump transactions where another account elects to pay for someone else's transaction

A fee bump transaction wraps a normal transaction with two extra pieces of information: the account which will be paying for the fee and the new fee amount the account is willing to pay. Consequently, the transaction hash of the fee bump transaction will differ from the hash of the fee bump's inner transaction.

A fee bump transaction envelope will contain the signature of the fee source account.

The fee bump transaction envelope will also contain the full contents of the inner transaction including the original source account, signatures, and fee of the inner transaction.

To support Fee Bump transactions this PR implements the following changes in Horizon:

  -  add `inner_transaction_hash`, `inner_signatures`, `inner_max_fee`,  and `fee_account` columns to the `history_transactions` table
  -  if a transaction **is** a fee bump transaction, `inner_transaction_hash` will be the hash of the inner transaction wrapped by the fee bump, `inner_signatures` will be the set of signatures for the inner transaction, `inner_max_fee` will be the max fee set in the inner transaction, and `fee_account` will be the account paying the fee on behalf of the inner transaction
  - Update ledger transaction processors because the way we extract operation results from the transaction results XDR object has changed
 - Include the fee account as a participant of the transaction in ingestion
 - Update `txsub` package in horizon so that a request to submit a normal transaction will respond with the transaction result from its corresponding fee bump transaction if the normal transaction was fee bumped
 - Update request handlers which look up information based on transaction hash (e.g. `GET /transactions/{hash}`) so that the `hash` parameter can match either the `transaction_hash` or `inner_transaction_hash` column from the `history_transactions` table. In other words, we should be able to look up a transaction by either its hash or its inner hash (in the case the transaction is a fee bump)

### Known limitations

We are punting on adding any new types of effects for fee bump transactions because Horizon effects are already missing cases like "Account Debited" for transaction fees (see https://github.com/stellar/go/issues/2175)

### CODE REVIEW

File: services/horizon/internal/db2/history/main.go [hunk 1]
```diff
@@ -585,7 +585,11 @@ type Transaction struct {
 	// non-null value. We need `NULL` to indicate successful transaction because
 	// otherwise all existing transactions would be interpreted as failed until
 	// ledger is reingested.
-	Successful *bool `db:"successful"`
```
Comment (2opremio) [inline services/horizon/internal/db2/history/main.go]: We should really enforce `gofmt -s` linting.

**Comment by 2opremio** [comment]
I've read CAP15 and gone through the code and haven't found anything alarming but I don't really understand everything yet.

**Comment by tamirms** [comment]
I want to highlight the changes to the Horizon transaction responses. In this PR, the hash and id in the response always matches the hash in the  `GET /transactions/{hash}` request. 

<br>


* If a `GET /transactions/{hash}` request queries for a normal (non fee bump) transaction the response from horizon will resemble the [current format](https://www.stellar.org/developers/horizon/reference/resources/transaction.html) . The only difference is that there will be an extra string field called `fee_account`. In this case, `fee_account` will always equal `source_account`. Because the source account of the transaction is paying for the fee.

* If a `GET /transactions/{hash}` request looks up a transaction using the fee bump tx hash (as opposed to the inner tx hash) the response from horizon will have the following form:

<br/>

| Attribute               | Type                     |                                                                                                                                |
|-------------------------|--------------------------|--------------------------------------------------------------------------------------------------------------------------------|
| **id**                     | string                   | This attribute is equal to the fee bump tx hash, not the inner tx hash. |
| paging_token            | string                   | This attribute has not changed from the current format. |
| successful              | bool                     | This attribute has not changed from the current format. |
| **hash**                    | string                   | The hash is equal to the fee bump tx hash, not the inner tx hash.                                        |
| ledger                  | number                  | This attribute has not changed from the current format. |
| created_at              | ISO8601 string           | This attribute has not changed from the current format. |
| **fee_account**          | string                   | This is a new field. It is set to account which is paying for the fee                                                                                                                               |
| source_account | string                   |  This is the source account of the inner tx                                                                                                                              |
| source_account_sequence | string                   |  This is the sequence number of the source account of the inner tx                                                                                                                              |
| **max_fee**                 | number                   | The the maximum fee the **fee account** was willing to pay for the fee bump tx.                                                                     |
| **fee_charged**             | number                   | The fee paid by the  **fee account**  when the transaction was applied to the ledger.                         |
| operation_count         | number                   | The number of operations that are contained within the **inner tx**.                                                           |
| envelope_xdr            | string                   | The XDR string for the **fee bump tx envelope**                                       |
| result_xdr              | string                   | The XDR string for the **fee bump** `TransactionResult`. Note that this XDR object also contains results from operations in the **inner tx**.                                         |
| result_meta_xdr         | string                   | The XDR string for `TransactionMeta` (note that the XDR definition hasn't changed for this field and that there is no concept of TransactionMeta applying to the inner transaction separately from the fee bump transaction).                                           |
| fee_meta_xdr            | string                   | The XDR string for `LedgerEntryChanges` xdr struct produced by taking fees for this transaction (note that the XDR definition hasn't changed for this field and that there is no reason to separate `LedgerEntryChanges` into inner tx changes and fee bump tx changes)               |
| memo_type               | string                   |    The memo type for the **inner tx**.                                                                                                                            |
| memo                    | string                   |    The memo for the **inner tx**.                                                                                                                            |
| **signatures**              | string[]                 | The list of signatures used to sign the **fee bump tx**                                                                           |
| valid_after             | RFC3339 date-time string |  Derived from the **inner tx** time bounds.                                                                                                                              |
| valid_before            | RFC3339 date-time string | Derived from the **inner tx** time bounds.                                                                                                                               |
| **fee_bump_transaction**            | Object with the following fields: `hash`, `signatures` | This object contains the max fee from the fee bump transaction, and a list of signatures from the fee bump transaction                                                                                                                               |
| **inner_transaction**            | Object with the following fields: `max_fee`, `hash`, `signatures` | This object contains the max fee from the inner transaction, the hash of the inner transaction, and a list of signatures from the inner transaction                                                                                                                               |

<br/>

* If a `GET /transactions/{hash}` request looks up a transaction using the inner tx hash (as opposed to the fee bump tx hash) the response from horizon will have the following form:

<br/>

| Attribute               | Type                     |                                                                                                                                |
|-------------------------|--------------------------|--------------------------------------------------------------------------------------------------------------------------------|
| **id**                      | string                   | This attribute is equal to the inner tx hash, not the fee bump tx hash. |
| paging_token            | string                   | This attribute has not changed from the current format. |
| successful              | bool                     | This attribute has not changed from the current format. |
| **hash**                    | string                   | The hash is equal to the inner tx hash, not the fee bump tx hash.                                        |
| ledger                  | number                  | This attribute has not changed from the current format. |
| created_at              | ISO8601 string           | This attribute has not changed from the current format. |
| **fee_account**          | string                   | This is a new field. It is set to account which is paying for the fee                                                                                                                               |
| source_account | string                   |  This is the source account of the inner tx                                                                                                                              |
| source_account_sequence | string                   |  This is the sequence number of the source account of the inner tx                                                                                                                              |
| **max_fee**                 | number                   | The the maximum fee the **fee account** was willing to pay for the fee bump tx.                                                                     |
| **fee_charged**             | number                   | The fee paid by the  **fee account**  when the transaction was applied to the ledger.                         |
| operation_count         | number                   | The number of operations that are contained within the **inner tx**.                                                           |
| envelope_xdr            | string                   | The XDR string for the **fee bump tx envelope**. Note that this XDR object also contains the inner tx envelope                                       |
| result_xdr              | string                   | The XDR string for the **fee bump** `TransactionResult`. Note that this XDR object also contains results from operations in the **inner tx**.                                         |
| result_meta_xdr         | string                   | The XDR string for `TransactionMeta`                                           |
| fee_meta_xdr            | string                   | The XDR string for `LedgerEntryChanges` xdr struct produced by taking fees for this transaction               |
| memo                    | string                   |    The memo for the **inner tx**.                                                                                                                            |
| **signatures**              | string[]                 | The list of signatures used to sign the **inner tx**                                                                           |
| valid_after             | RFC3339 date-time string |  Derived from the **inner tx** time bounds.                                                                                                                              |
| valid_before            | RFC3339 date-time string | Derived from the **inner tx** time bounds.                                                                                                                               |
| **fee_bump_transaction**            | Object with the following fields: `hash`, `signatures` | This object contains the max fee from the fee bump transaction, and a list of signatures from the fee bump transaction                                                                                                                               |
| **inner_transaction**            | Object with the following fields: `max_fee`, `hash`, `signatures` | This object contains the max fee from the inner transaction, the hash of the inner transaction, and a list of signatures from the inner transaction                                                                                                                               |

**Comment by tamirms** [comment]
When submitting a transaction the structure of the response remains the same:

```golang
type TransactionSuccess struct {
	Links struct {
		Transaction hal.Link `json:"transaction"`
	} `json:"_links"`
	Hash   string `json:"hash"`
	Ledger int32  `json:"ledger"`
	Env    string `json:"envelope_xdr"`
	Result string `json:"result_xdr"`
	Meta   string `json:"result_meta_xdr"`
}
```

However, if you submit a transaction and someone else fee bumps that transaction, then the response to your submit request will still have a `Hash` field which matches your request. However, the `Env` field will not match your request. The `Env` field will instead consist of the fee bump transaction envelope which wrapped your transaction.

File: services/horizon/internal/actions/transaction_test.go [hunk 2]
```diff
@@ -86,3 +88,92 @@ func TestLoadTransactionRecords(t *testing.T) {
...
+	tt.Assert.NotEqual(byOuterHash.Signatures, byInnerHash.Signatures)
+
+	tt.Assert.Equal(fixture.InnerHash, byInnerHash.Hash)
+	tt.Assert.Equal(fixture.InnerHash, byInnerHash.ID)
+	tt.Assert.Equal(
+		strings.Split(fixture.Transaction.InnerSignatureString, ","),
+		byInnerHash.Signatures,
+	)
+
+	byInnerHash.Hash = byOuterHash.Hash
```
Comment (abuiles) [inline services/horizon/internal/actions/transaction_test.go]: @tamirms is this testing something?

Comment (tamirms) [reply services/horizon/internal/actions/transaction_test.go]: thanks for pointing this out, I need to add `tt.Assert.Equal(byOuterHash, byInnerHash)`. I'll push a fix soon

**Comment by ire-and-curses** [comment]
> When submitting a transaction the structure of the response remains the same:
> 
> ```go
> type TransactionSuccess struct {
> 	Links struct {
> 		Transaction hal.Link `json:"transaction"`
> 	} `json:"_links"`
> 	Hash   string `json:"hash"`
> 	Ledger int32  `json:"ledger"`
> 	Env    string `json:"envelope_xdr"`
> 	Result string `json:"result_xdr"`
> 	Meta   string `json:"result_meta_xdr"`
> }
> ```
> 
> However, if you submit a transaction and someone else fee bumps that transaction, then the response to your submit request will still have a `Hash` field which matches your request. However, the `Env` field will not match your request. The `Env` field will instead consist of the fee bump transaction envelope which wrapped your transaction.

This is interesting - what was your rationale here?

File: services/horizon/internal/db2/history/transaction.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ func (t *Transaction) IsSuccessful() bool {
 func (q *Q) TransactionByHash(dest interface{}, hash string) error {
 	sql := selectTransaction.
 		Limit(1).
-		Where("ht.transaction_hash = ?", hash)
+		Where("ht.transaction_hash = ? OR ht.inner_transaction_hash = ?", hash, hash)
```
Comment (ire-and-curses) [inline services/horizon/internal/db2/history/transaction.go]: Are there performance implications to this for non-fee bump hash lookups? Or does this short circuit and eval the first part before looking for the `inner_transaction_hash`?

**Comment by tamirms** [comment]
@ire-and-curses 

>This is interesting - what was your rationale here?

I actually don't have a strong opinion here. I think it would be nice if there was some way to determine if your transaction has been fee bumped. If the `Env` field matches the the fee bump transaction envelope you could then infer that your transaction was fee bumped. Alternatively, we could make both the hash and the env match the fee bump transaction.

One idea I had was to actually include the full transaction response in `TransactionSuccess`. I imagine that a lot of clients end up doing a `GET /transactions/<hash>` response immediately after submitting a transaction so having the transaction response in `TransactionSuccess` would remove the need to do the follow up `GET /transactions/<hash>` request. Doing that might also address the underlying need for https://github.com/stellar/go/issues/1094

@ire-and-curses what do you think? do you have any preference between these possible solutions?

File: services/horizon/internal/db2/history/transaction.go (hunk 3 above)
Comment (tamirms) [reply services/horizon/internal/db2/history/transaction.go]: I believe that using an `OR` condition will not impact the performance of the query significantly. I used `explain analyze` on the horizon staging db to compare the following queries:

- `select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' OR id = 124082318146162688;`
- `select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' UNION select * from history_transactions where id = 124082318146162688;`
- `select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' UNION ALL select * from history_transactions where id = 124082318146162688;`

All these queries produce equivalent results. As a baseline measure, I used  `select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889';`

The results indicate seem to indicate that using the `OR` condition is the best way to write the query.

```
explain analyze verbose select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' OR id = 124082318146162688;

 Bitmap Heap Scan on public.history_transactions  (cost=3.29..5.32 rows=2 width=1648) (actual time=0.036..0.038 rows=2 loops=1)
   Output: transaction_hash, ledger_sequence, application_order, account, account_sequence, max_fee, operation_count, created_at, updated_at, id, tx_envelope, tx_result, tx_meta, tx_fee_meta, signatures, memo_type, memo, time_bounds, successful, fee_charged
   Recheck Cond: (((history_transactions.transaction_hash)::text = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889'::text) OR (history_transactions.id = '124082318146162688'::bigint))
   Heap Blocks: exact=2
   ->  BitmapOr  (cost=3.29..3.29 rows=2 width=0) (actual time=0.032..0.032 rows=0 loops=1)
         ->  Bitmap Index Scan on by_hash  (cost=0.00..1.71 rows=1 width=0) (actual time=0.021..0.021 rows=1 loops=1)
               Index Cond: ((history_transactions.transaction_hash)::text = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889'::text)
         ->  Bitmap Index Scan on index_history_transactions_on_id  (cost=0.00..1.58 rows=1 width=0) (actual time=0.010..0.010 rows=1 loops=1)
               Index Cond: (history_transactions.id = '124082318146162688'::bigint)
 Planning time: 0.089 ms
 Execution time: 0.067 ms
(11 rows)

explain analyse verbose select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' UNION select * from history_transactions where id = 124082318146162688;

 Unique  (cost=5.33..5.44 rows=2 width=1648) (actual time=0.065..0.069 rows=2 loops=1)
   Output: history_transactions.transaction_hash, history_transactions.ledger_sequence, history_transactions.application_order, history_transactions.account, history_transactions.account_sequence, history_transactions.max_fee, history_transactions.operation_count, history_transactions.created_at, history_transactions.updated_at, history_transactions.id, history_transactions.tx_envelope, history_transactions.tx_result, history_transactions.tx_meta, history_transactions.tx_fee_meta, history_transactions.signatures, history_transactions.memo_type, history_transactions.memo, history_transactions.time_bounds, history_transactions.successful, history_transactions.fee_charged
   ->  Sort  (cost=5.33..5.34 rows=2 width=1648) (actual time=0.063..0.064 rows=2 loops=1)
         Output: history_transactions.transaction_hash, history_transactions.ledger_sequence, history_transactions.application_order, history_transactions.account, history_transactions.account_sequence, history_transactions.max_fee, history_transactions.operation_count, history_transactions.created_at, history_transactions.updated_at, history_transactions.id, history_transactions.tx_envelope, history_transactions.tx_result, history_transactions.tx_meta, history_transactions.tx_fee_meta, history_transactions.signatures, history_transactions.memo_type, history_transactions.memo, history_transactions.time_bounds, history_transactions.successful, history_transactions.fee_charged
         Sort Key: history_transactions.transaction_hash, history_transactions.ledger_sequence, history_transactions.application_order, history_transactions.account, history_transactions.account_sequence, history_transactions.max_fee, history_transactions.operation_count, history_transactions.created_at, history_transactions.updated_at, history_transactions.id, history_transactions.tx_envelope, history_transactions.tx_result, history_transactions.tx_meta, history_transactions.tx_fee_meta, history_transactions.signatures, history_transactions.memo_type, history_transactions.memo, history_transactions.time_bounds, history_transactions.successful, history_transactions.fee_charged
         Sort Method: quicksort  Memory: 29kB
         ->  Append  (cost=0.70..5.32 rows=2 width=1648) (actual time=0.025..0.036 rows=2 loops=1)
               ->  Index Scan using by_hash on public.history_transactions  (cost=0.70..2.71 rows=1 width=1648) (actual time=0.024..0.025 rows=1 loops=1)
                     Output: history_transactions.transaction_hash, history_transactions.ledger_sequence, history_transactions.application_order, history_transactions.account, history_transactions.account_sequence, history_transactions.max_fee, history_transactions.operation_count, history_transactions.created_at, history_transactions.updated_at, history_transactions.id, history_transactions.tx_envelope, history_transactions.tx_result, history_transactions.tx_meta, history_transactions.tx_fee_meta, history_transactions.signatures, history_transactions.memo_type, history_transactions.memo, history_transactions.time_bounds, history_transactions.successful, history_transactions.fee_charged
                     Index Cond: ((history_transactions.transaction_hash)::text = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889'::text)
               ->  Index Scan using index_history_transactions_on_id on public.history_transactions history_transactions_1  (cost=0.57..2.59 rows=1 width=1648) (actual time=0.009..0.009 rows=1 loops=1)
                     Output: history_transactions_1.transaction_hash, history_transactions_1.ledger_sequence, history_transactions_1.application_order, history_transactions_1.account, history_transactions_1.account_sequence, history_transactions_1.max_fee, history_transactions_1.operation_count, history_transactions_1.created_at, history_transactions_1.updated_at, history_transactions_1.id, history_transactions_1.tx_envelope, history_transactions_1.tx_result, history_transactions_1.tx_meta, history_transactions_1.tx_fee_meta, history_transactions_1.signatures, history_transactions_1.memo_type, history_transactions_1.memo, history_transactions_1.time_bounds, history_transactions_1.successful, history_transactions_1.fee_charged
                     Index Cond: (history_transactions_1.id = '124082318146162688'::bigint)
 Planning time: 0.181 ms
 Execution time: 0.116 ms
(15 rows)

explain analyse verbose select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889' UNION ALL select * from history_transactions where id = 124082318146162688;

 Append  (cost=0.70..5.32 rows=2 width=1648) (actual time=0.022..0.032 rows=2 loops=1)
   ->  Index Scan using by_hash on public.history_transactions  (cost=0.70..2.71 rows=1 width=1648) (actual time=0.022..0.022 rows=1 loops=1)
         Output: history_transactions.transaction_hash, history_transactions.ledger_sequence, history_transactions.application_order, history_transactions.account, history_transactions.account_sequence, history_transactions.max_fee, history_transactions.operation_count, history_transactions.created_at, history_transactions.updated_at, history_transactions.id, history_transactions.tx_envelope, history_transactions.tx_result, history_transactions.tx_meta, history_transactions.tx_fee_meta, history_transactions.signatures, history_transactions.memo_type, history_transactions.memo, history_transactions.time_bounds, history_transactions.successful, history_transactions.fee_charged
         Index Cond: ((history_transactions.transaction_hash)::text = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889'::text)
   ->  Index Scan using index_history_transactions_on_id on public.history_transactions history_transactions_1  (cost=0.57..2.59 rows=1 width=1648) (actual time=0.008..0.009 rows=1 loops=1)
         Output: history_transactions_1.transaction_hash, history_transactions_1.ledger_sequence, history_transactions_1.application_order, history_transactions_1.account, history_transactions_1.account_sequence, history_transactions_1.max_fee, history_transactions_1.operation_count, history_transactions_1.created_at, history_transactions_1.updated_at, history_transactions_1.id, history_transactions_1.tx_envelope, history_transactions_1.tx_result, history_transactions_1.tx_meta, history_transactions_1.tx_fee_meta, history_transactions_1.signatures, history_transactions_1.memo_type, history_transactions_1.memo, history_transactions_1.time_bounds, history_transactions_1.successful, history_transactions_1.fee_charged
         Index Cond: (history_transactions_1.id = '124082318146162688'::bigint)
 Planning time: 0.164 ms
 Execution time: 0.066 ms
(9 rows)


explain analyse verbose select * from history_transactions where transaction_hash = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889';

 Index Scan using by_hash on public.history_transactions  (cost=0.70..2.71 rows=1 width=1648) (actual time=0.023..0.024 rows=1 loops=1)
   Output: transaction_hash, ledger_sequence, application_order, account, account_sequence, max_fee, operation_count, created_at, updated_at, id, tx_envelope, tx_result, tx_meta, tx_fee_meta, signatures, memo_type, memo, time_bounds, successful, fee_charged
   Index Cond: ((history_transactions.transaction_hash)::text = '3389e9f0f1a65f19736cacf544c2e825313e8447f569233bb8db39aa607c8889'::text)
 Planning time: 0.068 ms
 Execution time: 0.041 ms
(5 rows)
```

I will take another look at the performance once we deploy horizon v1.1.0 to staging

**Comment by ire-and-curses** [comment]
> @ire-and-curses
> 
> > This is interesting - what was your rationale here?
> 
> I actually don't have a strong opinion here. I think it would be nice if there was some way to determine if your transaction has been fee bumped. If the `Env` field matches the the fee bump transaction envelope you could then infer that your transaction was fee bumped. Alternatively, we could make both the hash and the env match the fee bump transaction.
> 
> One idea I had was to actually include the full transaction response in `TransactionSuccess`. I imagine that a lot of clients end up doing a `GET /transactions/<hash>` response immediately after submitting a transaction so having the transaction response in `TransactionSuccess` would remove the need to do the follow up `GET /transactions/<hash>` request. Doing that might also address the underlying need for #1094
> 
> @ire-and-curses what do you think? do you have any preference between these possible solutions?

I see. That is a nice feature, to be able to see that you were bumped.

I'm interested in the idea of returning the full transaction response. Do you think there are any performance implications? If not, then this seems like a win to me.

**Comment by tamirms** [comment]
> I'm interested in the idea of returning the full transaction response. Do you think there are any performance implications? If not, then this seems like a win to me.

we don't have to do any extra database queries because we have the full transaction row. txsub actually filters away properties from the transaction row to produce the `TransactionSuccess` struct

File: services/horizon/CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,8 @@ file. This project adheres to [Semantic Versioning](http://semver.org/).
 * Fix ask and bid price levels of GET /order_book when encountering non-canonical price values. The `limit` parameter is now respected and levels are coallesced properly. Also, `price_r` is now in canonical form.
 * Remove duplicate indexes: index_history_transactions_on_id, index_history_ledgers_on_id, exp_asset_stats_by_code, and asset_by_code
 * Remove asset_stats table which is no longer necessary
+* Full transaction details are included in the `POST /transactions` response. If you submit a transaction and it succeeds, the response will match the `GET /transactions/{hash}` response.
+* The following attributes are now included in the transaction resource: `fee_account` (the account which paid the transaction fees), `fee_bump_transaction` (only present in fee bump transactions),  `inner_transaction` (only present in fee bump transactions)
```
Comment (abuiles) [inline services/horizon/CHANGELOG.md]:
```suggestion
* Full transaction details are included in the `POST /transactions` response. If you submit a transaction and it succeeds, the response will match the `GET /transactions/{hash}` response ([#2406](https://github.com/stellar/go/pull/2406)).
* The following attributes are now included in the transaction resource: `fee_account` (the account which paid the transaction fees), `fee_bump_transaction` (only present in fee bump transactions),  `inner_transaction` (only present in fee bump transactions) ([#2406](https://github.com/stellar/go/pull/2406)).
```

**Pull request merged**